releasing/versioning_pre_nightly.md

```diff
@@ -0,0 +1,56 @@
+## Versioning in nightly and prerelease binaries
+
+Binary packages produced for prerelease and nightly builds has some
+particularities to stablish the priority among them nicely.
+
+### Version schemes
+
+**Prerelease** versioning scheme: `{upcoming_version}~pre{prerelease_version}`
+
+ * `upcoming_version:` upstream version target for current prerelease series
+ * `prerelease_version`: prerelease version number in the series
+
+**Nightly** use the following versioning scheme: `{current_released_version}+git{date}+${nightly_revision}r{hash}-{nightly_revision}`
+
+ * `current_released_version:` will be the latest version released available in
+   the changelog file of the corresponding -release repo. If the nightly is
+   used for an upcoming release (for example, gazebo10) then R-1.99.99-1
+   (gazebo10_9.99.99-1) form will be used until prereleases or final release.
```

Comment:
Should `9` be `1` from R-1.99.99-1?

```suggestion
   (gazebo10_1.99.99-1) form will be used until prereleases or final release.
```

Reply:
`9.99.99` is the version (X.Y.Z) used by nightlies to avoid to take precedence over `10.x.y` prereleases or releases. The revision number always appear after `-` in debian packaging so `1` in the example.

Reply:
I see, it might be good to add a statement on what the precedence is. From what I'm gathering it's stable > prerelease > nightly?

Reply:
almost, not always the case if we use nightlies as quick releases before doing a new stable one, then we use the trick of X.99.99. Let me know if these changes make sense to you 4fc6792

Reply:
The changes helped me better understand but there is one statement I don't quite get. See https://github.com/ignitionrobotics/docs/pull/140#discussion_r574697133 I'm not sure what "Another use case for nightlies is to server as continuous unstable releases when preparing a new version `X.Y.Z`." means

Reply:
The idea is that producing nightlies all the days will make the stable release easy since all the code and release metadata have been tested together while producing the nightlies so we avoid surprises at the time of the stable release. Does this make sense to you?